### PR TITLE
add confirmation modification of delete command

### PR DESCRIPTION
**Pull request by AllyW**

### CODE REVIEW

**Comment by AllyW** [comment]
image

File: src/aaz_dev/command/api/editor.py [hunk 1]
```diff
@@ -186,6 +186,8 @@ def editor_workspace_command(name, node_names, leaf_name):
             leaf = manager.update_command_tree_leaf_stage(*leaf.names, stage=data['stage'])
         if 'examples' in data:
             leaf = manager.update_command_tree_leaf_examples(*leaf.names, examples=data['examples'])
+        if 'confirmation' in data and leaf.confirmation != data['confirmation']:
+            leaf = manager.update_command_tree_leaf_confirmation(*leaf.names, confirmation=data['confirmation'])
```
Comment (kairu-ms) [inline src/aaz_dev/command/api/editor.py:190]: I think there's no need to verify `leaf.confirmation != data['confirmation']`

File: src/aaz_dev/command/model/editor/_workspace.py [hunk 2]
```diff
@@ -13,6 +13,7 @@ class CMDCommandTreeLeaf(Model):
     help = ModelType(CMDHelp)
     resources = ListType(ModelType(CMDResource), min_size=1)  # the azure resources used in this command
     examples = ListType(ModelType(CMDCommandExample))
+    confirmation = CMDConfirmation()
```
Comment (kairu-ms) [inline src/aaz_dev/command/model/editor/_workspace.py:16]: Not recommand to duplicate the definition in two place. The confirmation property is defined in [[src/aaz_dev/command/model/configuration/_command.py](https://github.com/Azure/aaz-dev-tools/pull/141/files#diff-dac2eec0f1da7589f603c138da44f9a420568fb245cb38718a54d751cb18b49f)](https://github.com/Azure/aaz-dev-tools/blob/c3862438f84f7a4e8b7a31528e8bced751fd85aa/src/aaz_dev/command/model/configuration/_command.py#L18-L37)

File: src/aaz_dev/swagger/controller/command_generator.py [hunk 3]
```diff
@@ -144,9 +144,14 @@ def create_draft_command_group(self, resource,
...
         return resource.version
 
+    @staticmethod
+    def generate_command_confirmation(resource):
+        return resource.confirmation
+
     def generate_command(self, path_item, resource, cmd_builder):
         command = CMDCommand()
         command.version = self.generate_command_version(resource)
+        command.confirmation = self.generate_command_confirmation(resource)
```
Comment (kairu-ms) [inline src/aaz_dev/swagger/controller/command_generator.py:154]: I don't think swagger resource has confirmation property.

**Review by kairu-ms — CHANGES_REQUESTED** [review]
The code will not work. Not qualified to merge.

File: src/aaz_dev/command/controller/workspace_manager.py [hunk 4]
```diff
@@ -356,6 +356,16 @@ def update_command_tree_leaf_stage(self, *leaf_names, stage):
...
 
+    def update_command_tree_leaf_confirmation(self, *leaf_names, confirmation):
+        leaf = self.find_command_tree_leaf(*leaf_names)
+        if not leaf:
+            raise exceptions.ResourceNotFind(f"Command Tree leaf not found: '{' '.join(leaf_names)}'")
+
+        if leaf.confirmation == confirmation:
+            return
+        leaf.confirmation = confirmation
+        return leaf
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/workspace_manager.py:367]: it seems this function is useless

File: src/web/src/views/workspace/WSEditor.tsx [hunk 5]
```diff
@@ -322,16 +325,23 @@ class WSEditor extends React.Component<WSEditorProps, WSEditorState> {
...
+            axios.get(leafUrl).then(res => {
+                const cmd = DecodeResponseCommand(res.data);
+                this.setState(preState => {
+                    let selected = preState.commandMap[nodeId];
+                    selected.confirmation = cmd.confirmation;
+                    return {
+                        ...preState,
+                        selected: selected,
+                    }
+                })
```
Comment (kairu-ms) [inline src/web/src/views/workspace/WSEditor.tsx:342]: It's bad to add confirmation in command tree and make request here.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**